components/payments/internal/connectors/plugins/public/stripe/accounts.go

```diff
@@ -0,0 +1,91 @@
+package stripe
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/formancehq/payments/internal/connectors/plugins/currency"
+	"github.com/formancehq/payments/internal/models"
+)
+
+var (
+	rootAccountReference = "root"
+)
+
+// root account reference is internal so we don't pass it to Stripe API clients
+func resolveAccount(ref *string) *string {
+	if *ref == rootAccountReference {
+		return nil
+	}
+	return ref
+}
+
+type AccountsState struct {
+	InitFinished bool   `json:"init_finished"`
+	LastID       string `json:"lastID,omitempty"`
+}
+
+func (p *Plugin) fetchNextAccounts(ctx context.Context, req models.FetchNextAccountsRequest) (models.FetchNextAccountsResponse, error) {
+	var oldState AccountsState
+	if req.State != nil {
+		if err := json.Unmarshal(req.State, &oldState); err != nil {
+			return models.FetchNextAccountsResponse{}, err
+		}
+	}
+
+	accounts := make([]models.PSPAccount, 0, req.PageSize)
+	if !oldState.InitFinished {
+		// create a root account if this is the first time this is being run
+		accounts = append(accounts, models.PSPAccount{
+			Name:      &rootAccountReference,
+			Reference: rootAccountReference,
+			CreatedAt: time.Now().UTC(),
+			Raw:       json.RawMessage("{}"),
+			Metadata:  map[string]string{},
+		})
+		oldState.InitFinished = true
+	}
+
+	needed := req.PageSize - len(accounts)
```

Comment:
The PageSize request parameters can be set by the user at connectors creation, and stripe "limit" query param is only valid if between 1 and 100. So if the users puts something > 100, all calls will return an error !

Reply:
Created a ticke for this here: https://linear.app/formance/issue/ENG-1420/featpayments-validate-page-size-in-connectors